### ISSUE TITLE
GWT compilation error in Eclipse with GWT 3.0 plugin

### ISSUE THREAD

**Issue by jiny4clt** [gwt-plugins/gwt-eclipse-plugin#470]
Hello,

Please help with this weird GWT compilation error encountered in my development environment recently.
The GWT compilation process used to work in my laptop until recently it stopped compiling the GWT code with the following error.
There is no detailed error message can be found when I clicked on the "Details" button.

I recalled that recently I installed Docker Desktop and different releases of Tomcat 9 server for local Tomcat deployment. After I uninstalled Docker Desktop, the compilation error still persists.

This is a really big headache and cause issues to my development and deployment process. I've tried to reinstall the Eclipse and GWT 3 plugin and used different workspaces, but still can't find any solution. I posted my issue to GWT user group and the staff suggested me to report this problem in this place. Please help shed some lights on this issue.

Eclipse version: 2020-06 (4.16.0)
GWT 2.9
JDK8/JDK11

Error: Could not find or load main class com.google.gwt.dev.Compiler
Caused by: java.lang.ClassNotFoundException: com.google.gwt.dev.Compiler

Please see below for the GWT related configurations in Eclipse:
![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/1cc8ef9e-bd5f-45db-ba9b-02ef481fe7bc)

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/961bb4d8-5420-4a9e-89b7-35d0eb723adb)

Your help is much appreciated!

thanks,
Jenny

**Comment by jiny4clt** [comment]
In addition, I've tried to install other version of Eclipse such as Eclipse 4.18 (2020-12) with a new workspace. After I installed GWT plugin 3 from market place and tried to create a sample gwt web application project 
as described below in gwt site, I got the error " **Invocation of com.google.gwt.user.tools.WebAppCreator** failed. See the error log for more details."  It seems to me this current market place gwt plugin didn't work as I didn't use our project at all
and followed the tutorial below.

https://www.gwtproject.org/usingeclipse.html

Is there a way to get a working GWT plugin for Eclipse which can work with JDK 11?

**Comment by niloc132** [comment]
I wonder if you can share the actual sample project, the entire log, and a listing of any/all plugins installed? It is quite difficult to debug screenshots, and as you noted, this works on other computers (your old laptop I think you mentioned, with the same install of eclipse? and certainly on our computers that we use to develop, test, and deploy the plugin).

As this is only failing on your new computer, and we haven't heard this reported from anyone else, it sounds like it might be specific to your environment, at least until we have enough details to actually try something.

**Comment by jhillbounce** [comment]
@jiny4clt  Sorry that is happening.  If it helps at all - here are two confirmed versions that work well. 

Newer Eclipse and Newer GWT Plugin
Eclipse:
Version: 2023-03 (4.27.0)
Build id: 20230309-1520

GWT Eclipse Plugin	
4.0.0.202302020242

Older Eclipse and Older GWT Plugin
Version: 2020-12 (4.18.0)
Build id: 20201210-1552

GWT Eclipse Plugin
3.0.0.201710131939

**Comment by jiny4clt** [comment]
> I wonder if you can share the actual sample project, the entire log, and a listing of any/all plugins installed? It is quite difficult to debug screenshots, and as you noted, this works on other computers (your old laptop I think you mentioned, with the same install of eclipse? and certainly on our computers that we use to develop, test, and deploy the plugin).
> 
> As this is only failing on your new computer, and we haven't heard this reported from anyone else, it sounds like it might be specific to your environment, at least until we have enough details to actually try something.

Colin, 

As I mentioned in the above post, I was trying to follow the instructions on the GWT site below for Using Eclipse to create a new GWT Web Application Project, and the error "Invocation of com.google.gwt.user.tools.WebAppCreator"  came up. Actually, the sample GWT project has not been created. I tried to find the Eclipse log file, but I haven't found it yet. Please advise about where to find the error related stack track from Eclipse log file for this error.

**https://www.gwtproject.org/usingeclipse.html**

The GWT Eclipse Plugin downloaded from Market place is the version of 3.0.0.201710131939 and Eclipse version is 4.18 (2020-12)
  GWT Eclipse Plugin	3.0.0.201710131939	com.gwtplugins.eclipse.suite.v3.feature.feature.group	GWT Eclipse Plugin

thanks, 
Jenny

**Comment by jiny4clt** [comment]
> @jiny4clt Sorry that is happening. If it helps at all - here are two confirmed versions that work well.
> 
> Newer Eclipse and Newer GWT Plugin Eclipse: Version: 2023-03 (4.27.0) Build id: 20230309-1520
> 
> GWT Eclipse Plugin 4.0.0.202302020242
> 
> Older Eclipse and Older GWT Plugin Version: 2020-12 (4.18.0) Build id: 20201210-1552
> 
> GWT Eclipse Plugin 3.0.0.201710131939



> 



> 

Thanks for the information.

I think my new laptop is using the one you provided for the older Eclipse and GWT plugin, however, I got the error
" "Invocation of com.google.gwt.user.tools.WebAppCreator" when I just tried to create a new GWT web application project by following GWT tutorial. Any way to know what the error means? I also tried to copy Eclipse from my old laptop to the new one and also got the same error when trying to create a sample gwt project. I think if this error persists, it will also cause GWT compilation error for our GWT project, therefore, I didn't even import our project to this new workspace when trying to use the GWT plugin. Recently, I installed some software such as Docker Desktop and Tomcat 9 server in my new laptop. Did you think this will affect GWT plugin?

Currently, I have several different versions of Eclipse installed in my new laptop. Any possible conflicts for multiple Eclipse installed? It used to work well if I have multiple versions of Eclipse installed in my old laptop.

Your help is much appreciated,
Jenny
GWT Eclipse Plugin 3.0.0.201710131939 com.gwtplugins.eclipse.suite.v3.feature.feature.group GWT Eclipse Plugin


Older Eclipse and Older GWT Plugin
Version: 2020-12 (4.18.0)
Build id: 20201210-1552

GWT Eclipse Plugin
3.0.0.201710131939

**Comment by jhillbounce** [comment]
Hopefully this helps I was able to download the latest Eclipse and GWT plugin, and everything is working as expected for the starter "MyWebApp" GWT Project.

I downloaded latest Eclipse 2023-06 (https://www.eclipse.org/downloads/packages/release/kepler/sr2/eclipse-ide-java-ee-developers)

Selected "Eclipse IDE for Enterprise Java and Web Developers" option

Using Open JDK 19

Used the "install" button from this link https://marketplace.eclipse.org/content/gwt-plugin
To install GWT Eclipse Plugin 4.0.0

Restarted Eclipse

File...New...Other...GWT Web Application Project

Used Values for Project and Package as on website 
https://www.gwtproject.org/usingeclipse.html

Needed to select a GWT SDK...

Downloaded 2.10 from here:
https://www.gwtproject.org/download.html

Unzip GWT SDK on computer

Click "Configured SDK", clicked "Add" and selected the unzipped location of the downloaded GWT SDK 2.10

Compiled Example Project Successfully via the GWT Eclipse Plugin
![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/27926160/c8234607-33fd-400d-971f-de2710e9a73c)

Run the GWT Project
Right Click Project -> GWT->GWT Development Mode with Jetty...

Console Output Shows...
The code server is ready at http://127.0.0.1:9876/
Code server started in 1.918 s ms
[ERROR] jreLeakPrevention.gcDaemonFail
java.lang.ClassNotFoundException: sun.misc.GC
(I suspect this is that I'm using too new a version of the JDK - but GWT Still runs)

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/27926160/88e8fda9-2644-4a86-8c09-693161e9927e)


> Currently, I have several different versions of Eclipse installed in my new laptop. Any possible conflicts for multiple Eclipse installed? It used to work well if I have multiple versions of Eclipse installed in my old laptop.

I now have 3 versions of Eclipse and GWT installed on the same computer.  I do not think that is the issue.

Also looking at the classpath for ARIES.html - I am unsure how the GWT SDK was configured - but I have never seen the "persisted container".  All my projects simply show GWT SDK

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/27926160/03f94969-0640-47e7-b1df-87c03a92251a)

Here is what my GWT Settings Look Like In Eclipse.  (Note I always download the GWT SDK separately from the Eclipse plugin and link it here)
![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/27926160/c6cc8f83-26e8-422b-8d51-d0428735e9ee)

**Comment by jiny4clt** [comment]
Thanks very much for posting your steps for the installation and configuration. 

After I did exactly what you suggested for these steps, it still came up with the same Invocation error of WebAppCreator. I have no idea where to find the error log for more details and why this error always came up. I think I have tried so many ways trying to resolve this problem, but no luck yet. I assume there must something wrong for my new laptop. 

Could you point me about how to find the error log for this error? Or have any clue for this Invocation error of WebAppCreator for GWT plugin?

By checking Error log from Eclipse, it only said this is an Internal Error.
 
It's so time consuming trying to resolve this kind of error. If nothing works, I may have to have this new laptop reset and get everything installed and configured again. 

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/546be626-9528-4711-abc5-2c0f31efcede)

Error:
![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/88498ffe-5d3c-466b-ba05-ab2c485c60ef)

My New Eclipse and GWT Settings;
![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/eb4ff4fa-c9e5-4492-bae9-1b418e2e1367)

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/43055d62-6f5a-4cfc-8e25-fe8568abc172)

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/284f60d1-634f-4e58-ac50-5171595dafd9)

![image](https://github.com/gwt-plugins/gwt-eclipse-plugin/assets/138822110/b2266034-dda2-41b4-bac8-5d3c358481d2)

Anyway, thanks very much for you guys help!
Jenny

**Comment by jhillbounce** [comment]
You may want to check some basic Java settings as well  - do you have JAVA_HOME setup?  Java in your system path?

Are you able to build the samples on the command line with ant?

see:gwt-2.10.0\samples\build.xml

**Comment by jiny4clt** [comment]
Thanks very much for your help!

Finally, I've made it work to use GWT plugin in my eclipse to compile the GWT project. I came across the following post when searching for the error related to webappcreator. 

After I removed the "classpath" system variable from my new laptop and move the GWT SDK to the top of the build path, the GWT Compiler starts working to compile the GWT project. 

https://topic.alibabacloud.com/a/exception-invocation-of-font-colorredcomfont-font-colorredgooglefont-gwt-user-font-colorredtoolsfont-webappcreator-failed-see-the-error-log-for-more-details_8_8_31771273.html

thanks,
Jenny
